src/scratchOrgSettingsGenerator.ts

```diff
@@ -99,7 +86,8 @@ export default class SettingsGenerator {
   public async deploySettingsViaFolder(scratchOrg: Org, apiVersion: string): Promise<void> {
     const username = scratchOrg.getUsername();
     const logger = await Logger.child('deploySettingsViaFolder');
-    this.createPackageXml(apiVersion);
+    // this.createPackageXml(apiVersion);
+    await this.writePackageFile(this.allRecordTypes, this.allbusinessProcesses, this.packageFilePath, apiVersion);
```

Comment:
since `writePackageFile` is a method on the class, you could use the `this.` instance props instead of passing them in.

It might be cleaner to put those in an exported function (could be a different file) that would take inputs and return an object.

Then, your tests could pass a few permutations (with/without record types, BP, objectSettings, etc) and verify that the js objects for `CustomObject` and `package.xml` comes back as expected.

That'd be a really clean input/output function test that would cover most of the complexity since the test changes lost all the assertions about the xml contents